swaybar/ipc.c

```diff
@@ -322,8 +335,11 @@ void ipc_bar_init(struct bar *bar, const char *bar_id) {
 	}
 	free(res);
 	json_object_put(outputs);
-
-	const char *subscribe_json = "[ \"workspace\", \"mode\" ]";
+	// Will need to be redone for display mode toggling compatibility
+	// Will require setting bar->config->hidden_state to "show" as default
+	const char *subscribe_json = strcmp(bar->config->display_mode, "hide") == 0 ?
+		"[ \"workspace\", \"mode\", \"modifier\" ]"
+		: "[ \"workspace\", \"mode\" ]";
```

Comment:
Bleh. That's going to be annoying. Alright, here's the approach we should take: subscribe to only what we need, then add more subscriptions later if necessary as features are toggled on/off. We don't have to unsubscribe but we also don't have to subscribe until it's necessary.